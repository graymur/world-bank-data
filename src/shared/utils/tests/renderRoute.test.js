/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import renderRoute from '../renderRoute';

const config = {
	path: '/path',
	exact: true,
	component: () => (<h1>I'm an H1 tag</h1>)
};

test('Renders route if it matches url', () => {
	const component = renderer.create(
		<StaticRouter location="/path" context={{}}>
			{renderRoute(config)}
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
test('Renders nothing if rout does not match url', () => {
	const component = renderer.create(
		<StaticRouter location="/other-path" context={{}}>
			{renderRoute(config)}
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
