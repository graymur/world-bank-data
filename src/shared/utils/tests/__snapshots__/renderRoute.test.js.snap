// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders nothing if rout does not match url 1`] = `null`;

exports[`Renders route if it matches url 1`] = `
<h1>
  I'm an H1 tag
</h1>
`;
